### PR TITLE
Add environment identification tag in the header (409424)

### PR DESCRIPTION
**Pull request by KiranVarma02**
Adds an environment tag to the header

Production tag added for reference but will be hidden on **forms-runner**

<br>

Header tag development

Header tag performance test

Header tag production

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -27,6 +27,11 @@ export const config = convict({
     env: 'PORT'
   },
   env: {
+    format: ['development', 'test', 'pref test', 'production'],
```
Comment (colinrotherham) [inline src/config/index.ts:30]: Watch out for the `pref test` typo

Think we should match them to the actual env names? [See comment](https://github.com/DEFRA/forms-designer/pull/483/files#r1825666648)

```suggestion
    format: ['dev', 'test', 'perf-test', 'prod'],
```

File: src/server/views/layout.html [hunk 2]
```diff
@@ -45,7 +44,7 @@
       </a>
     
     {% if envTag %}
```
Comment (colinrotherham) [inline src/server/views/layout.html:46]: Do we need `envTag` everywhere?

```suggestion
    {% if config.env !== "prod" %}
```

File: src/server/views/tag-env/template.njk [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    {% case "production" %}
+      {% set classes = "govuk-tag--red" %}
+    {% default %}
+      {% set classes = "govuk-tag--grey" %}
+  {% endswitch -%}
+
+  {{ govukTag({
+    text: env | capitalize,
+    classes: classes
+  }) }}
```
Comment (colinrotherham) [inline src/server/views/tag-env/template.njk:15]:
```suggestion
  {% case "test" %}
    {% set classes = "govuk-tag--yellow" %}
  {% case "production" %}
    {% set classes = "govuk-tag--red" %}
  {% default %}
    {% set classes = "govuk-tag--grey" %}
{% endswitch -%}

{{ govukTag({
  text: env | capitalize,
  classes: classes
}) }}
```

File: src/config/index.ts (hunk 1 above)
Comment (KiranVarma02) [reply src/config/index.ts:30]: corrected typo. I will wait for @alexluckett to confirm regarding env names, same for designer PR

File: src/server/views/tag-env/template.njk (hunk 3 above)
Comment (colinrotherham) [reply src/server/views/tag-env/template.njk]: Same 😭

Comment (KiranVarma02) [reply src/server/views/tag-env/template.njk]: same? are you sure? did you pull latest?

Comment (colinrotherham) [reply src/server/views/tag-env/template.njk]: Ahh sorry, no I mean I also wish we had a linter for it! 😅

**Comment by KiranVarma02** [comment]
is there something we can do about sonar major issue? Its same in designer. I guess this can be ignored?

**Review by alexluckett — APPROVED** [review]

**Pull request merged**